Fix worklet referrer tests to match spec

domfarolino opened the pull request:
This PR fixes worklet referrer tests to match the spec. Basically this reverts a good bit of https://github.com/web-platform-tests/wpt/commit/50b0943d82d36bb92977002bebd5a7990e0ed388#diff-f5dd3dfe0ce7dd53814287bc30b9a197 (the WPTs bit of https://chromium-review.googlesource.com/c/chromium/src/+/1111743) to ensure that we actually do test that the referrer policy of the ["outer settings object"](https://drafts.css-houdini.org/worklets/#dom-worklet-addmodule) is honored when fetching worklet scripts. Related PR for module worker referrer tests: https://github.com/web-platform-tests/wpt/pull/12321.

/cc @nhiroki @nyaxt @hiroshige-g (Note: when this is merged and auto-rolls into Blink, I'll submit my CL that allows Blink to conform to these tests).

domfarolino committed:
Fix worklet referrer tests to match spec

domfarolino committed:
Add extra newline for readability

nyaxt commented on worklets/resources/referrer-checker.py:
```diff
@@ -2,14 +2,23 @@
+        if referrer == expected_referrer:
+            return (200, response_headers, "")
+        # The expected referrer doesn't contain query params for simplification,
+        # so we check the referrer by startswith() here.
```
Optional (since this isn't really introduced by this PR). I'm not a big fan of this logic. How about gating this on request.GET.first("expected_referrer_prefix")?

domfarolino committed:
Remove complex referrer-checker logic

domfarolino commented:
Ok so the reason for that logic was that the referrer header might contain query params and the expected referrer does not. What I've done is made the `referrer` variable equal only to the portion of the referrer header without query params.

When we compare it later on, we don't have to do any "`expected_referrer + ?`"ing. This is only relevant to the (same-origin referrer policy && same-origin request) case, so I've simplified the logic there. This has no effect on other cases, since `.split("?")[0]` is safe when "?" is not present.
Merging; if any additional feedback on this change please LMK.
